domain/consensus/timelock_CSV_test.go

```diff
@@ -0,0 +1,524 @@
+		if err != nil {
+			t.Fatalf("Error in createTransactionWithLockedOutput: %v", err)
+		}
+		// BlockE contains the locked output (locked by CSV).
+		// This block should be valid since CSV script locked only the output.
+		blockEHash, _, err := testConsensus.AddBlock([]*externalapi.DomainHash{blockDHash}, nil,
+			[]*externalapi.DomainTransaction{transactionWithLockedOutput})
+		if err != nil {
+			t.Fatalf("Error creating blockE: %v", err)
+		}
+		// bit 62 of sequence defines if it's conditioned by DAA score(set to 0) or by time (set to 1).
+		sequenceTypeFlag := 0
+		// Create a transaction that tries to spend the locked output.
+		transactionThatSpentTheLockedOutput, err := createTransactionThatSpentTheLockedOutputRelativeLock(transactionWithLockedOutput,
+			fees, redeemScriptCSV, numOfDAAScoreToWait, sequenceTypeFlag, blockEHash, &testConsensus)
+		if err != nil {
+			t.Fatalf("Error creating transactionThatSpentTheLockedOutput: %v", err)
+		}
+		// Add a block that contains a transaction that spends the locked output before the time, and therefore should be failed.
+		_, _, err = testConsensus.AddBlock([]*externalapi.DomainHash{blockEHash}, nil,
+			[]*externalapi.DomainTransaction{transactionThatSpentTheLockedOutput})
+		if err == nil || !errors.Is(err, ruleerrors.ErrUnfinalizedTx) {
+			t.Fatalf("Expected block to be invalid with err: %v, instead found: %v", ruleerrors.ErrUnfinalizedTx, err)
+		}
+		// Adds blocks until it reaches the DAA score target, so the locked output will be released.
+		tipHash := blockEHash
+		stagingArea := model.NewStagingArea()
+		blockEDAAScore, err := testConsensus.DAABlocksStore().DAAScore(testConsensus.DatabaseContext(), stagingArea, blockEHash)
+		if err != nil {
+			t.Fatalf("Failed getting DAA score of blockE: %v", err)
+		}
+		targetDAAScore := blockEDAAScore + numOfDAAScoreToWait
+		currentDAAScore, err := testConsensus.DAABlocksStore().DAAScore(testConsensus.DatabaseContext(), stagingArea, tipHash)
+		if err != nil {
+			t.Fatalf("Failed getting DAA score: %v", err)
+		}
+		for currentDAAScore <= targetDAAScore {
+			tipHash, _, err = testConsensus.AddBlock([]*externalapi.DomainHash{tipHash}, nil, nil)
+			if err != nil {
+				t.Fatalf("Error creating a tip: %v", err)
+			}
+			currentDAAScore, err = testConsensus.DAABlocksStore().DAAScore(testConsensus.DatabaseContext(), stagingArea, tipHash)
+			if err != nil {
+				t.Fatalf("Failed getting DAA score: %v", err)
+			}
+		}
+		// Tries to spend the output that should be no longer locked.
+		validBlock, _, err := testConsensus.AddBlock([]*externalapi.DomainHash{tipHash}, nil,
+			[]*externalapi.DomainTransaction{transactionThatSpentTheLockedOutput})
+		if err != nil {
+			t.Fatalf("The block should be valid since the output is not locked anymore. but got an error: %v", err)
+		}
+		validBlockStatus, err := testConsensus.BlockStatusStore().Get(testConsensus.DatabaseContext(), stagingArea,
```

Comment:
It's not valid yet. It's only expected to be valid.
Please rename to `actualBlockStatus` or just `blockStatus`